src/GenAPI/Microsoft.DotNet.GenAPI/SyntaxGeneratorExtensions.cs

```diff
@@ -46,10 +43,17 @@ public static SyntaxNode DeclarationExt(this SyntaxGenerator syntaxGenerator, IS
                 {
                     INamedTypeSymbol? baseType = method.ContainingType.BaseType;
                     // If the base type does not have default constructor.
-                    if (baseType != null && !baseType.Constructors.IsEmpty && baseType.Constructors.Where(c => c.Parameters.IsEmpty).Count() == 0)
+                    if (baseType != null && !baseType.Constructors.IsEmpty && baseType.Constructors.All(c => !c.Parameters.IsEmpty))
                     {
-                        ConstructorDeclarationSyntax declaration = (ConstructorDeclarationSyntax)syntaxGenerator.Declaration(method);
-                        return declaration.WithInitializer(GenerateBaseConstructorInitializer(baseType.Constructors));
+                        IOrderedEnumerable<IMethodSymbol> baseTypeConstructors = baseType.Constructors
+                            .Where(c => !c.GetAttributes().Any(a => a.IsObsoleteWithUsageTreatedAsCompilationError()))
```

Comment:
same here: Use All instead of !Any for readability:

```suggestion
                            .Where(c => c.GetAttributes().All(a => !a.IsObsoleteWithUsageTreatedAsCompilationError()))
```

Reply:
My expectation is that `Any` returns false when the first element found as a result in best case you don't need to traverse the whole list. On the other hand `All` will always traverse the list.

Reply:
> On the other hand All will always traverse the list.

That would be a very inefficient implementation of All. It returns as soon as it finds an element that doesn't match: https://source.dot.net/#System.Linq/System/Linq/AnyAll.cs,62

`All(a => !a.IsObsoleteWithUsageTreatedAsCompilationError())` is the just the more readable statement in comparison to `!Any(a => a.IsObsoleteWithUsageTreatedAsCompilationError())`.

Reply:
updated